File: core/usecases/src/main/java/com/mytaxi/common/usecases/di/ActivityUseCasesModule.java
package com.mytaxi.common.usecases.di;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import de.gfred.feature.one.MainActivity;
import de.gfred.feature.one.di.MainActivityComponent;


//@Module(includes = MainActivityBuilderModule.class)
//public abstract class ActivityUseCasesModule {
//}

@Module(subcomponents = MainActivityComponent.class)
public abstract class ActivityUseCasesModule {
    @ContributesAndroidInjector
    public abstract MainActivity contributeMainActivity();
}

File: core/usecases/src/main/java/com/mytaxi/common/usecases/di/ActivityUseCasesModule.kt
package com.mytaxi.common.usecases.di

import dagger.Module
import dagger.android.ContributesAndroidInjector
import de.gfred.feature.one.MainActivity

@Module
abstract class ActivityUseCasesModule {
    @ContributesAndroidInjector
    abstract fun contributeMainActivity(): MainActivity
}